tests/components/smartthings/test_light.py

```diff
@@ -0,0 +1,292 @@
+    broker = DeviceBroker(hass, devices, '')
+    config_entry = ConfigEntry("1", DOMAIN, "Test", {},
+                               SOURCE_USER, CONN_CLASS_CLOUD_PUSH)
+    hass.data[DOMAIN] = {
+        DATA_BROKERS: {
+            config_entry.entry_id: broker
+        }
+    }
+    await hass.config_entries.async_forward_entry_setup(config_entry, 'light')
+    await hass.async_block_till_done()
+    return config_entry
+
+
+async def test_async_setup_platform():
+    """Test setup platform does nothing (it uses config entries)."""
+    await light.async_setup_platform(None, None, None)
+
+
+def test_is_light(device_factory, light_devices):
+    """Test lights are correctly identified."""
+    non_lights = [
+        device_factory('Unknown', ['Unknown']),
+        device_factory("Fan 1",
+                       [Capability.switch, Capability.switch_level,
+                        Capability.fan_speed]),
+        device_factory("Switch 1", [Capability.switch]),
+        device_factory("Can't be turned off",
+                       [Capability.switch_level, Capability.color_control,
+                        Capability.color_temperature])
+    ]
+
+    for device in light_devices:
+        assert light.is_light(device), device.name
+    for device in non_lights:
+        assert not light.is_light(device), device.name
+
+
+async def test_entity_state(hass, light_devices):
+    """Tests the state attributes properly match the light types."""
+    await _setup_platform(hass, *light_devices)
+
+    # Dimmer 1
+    state = hass.states.get('light.dimmer_1')
+    assert state.state == 'on'
+    assert state.attributes[ATTR_SUPPORTED_FEATURES] == \
+        SUPPORT_BRIGHTNESS | SUPPORT_TRANSITION
+    assert state.attributes[ATTR_BRIGHTNESS] == 255
+
+    # Color Dimmer 1
+    state = hass.states.get('light.color_dimmer_1')
+    assert state.state == 'off'
+    assert state.attributes[ATTR_SUPPORTED_FEATURES] == \
+        SUPPORT_BRIGHTNESS | SUPPORT_TRANSITION | SUPPORT_COLOR
+
+    # Color Dimmer 2
+    state = hass.states.get('light.color_dimmer_2')
+    assert state.state == 'on'
+    assert state.attributes[ATTR_SUPPORTED_FEATURES] == \
+        SUPPORT_BRIGHTNESS | SUPPORT_TRANSITION | SUPPORT_COLOR | \
+        SUPPORT_COLOR_TEMP
+    assert state.attributes[ATTR_BRIGHTNESS] == 255
+    assert state.attributes[ATTR_HS_COLOR] == (273.6, 55.0)
+    assert state.attributes[ATTR_COLOR_TEMP] == 222
+
+
+async def test_entity_and_device_attributes(hass, device_factory):
+    """Test the attributes of the entity are correct."""
+    # Arrange
+    device = device_factory(
+        "Light 1", [Capability.switch, Capability.switch_level])
+    entity_registry = await hass.helpers.entity_registry.async_get_registry()
+    device_registry = await hass.helpers.device_registry.async_get_registry()
+    # Act
+    await _setup_platform(hass, device)
+    # Assert
+    entity = entity_registry.async_get("light.light_1")
```

Comment:
This is an entry for an entity we're getting. Maybe clarify name.

Reply:
Changed here and in the switch tests!